subprojects/integ-test/src/crossVersionTest/groovy/org/gradle/integtests/AbstractTaskSubclassingBinaryCompatibilityCrossVersionSpec.groovy

```diff
@@ -224,9 +217,13 @@ apply plugin: SomePlugin
 
             task t(type: SubclassTask)
         """
+    }
 
-        then:
-        version previous withTasks 'assemble' inDirectory(file("producer")) run()
-        version current requireDaemon() requireIsolatedDaemons() withTasks 't' run()
+    protected static String declareTaskClass(GradleVersion targetVersion) {
+        if (targetVersion >= GradleVersion.version("7.0")) {
```

Comment:
💬 Version 7.0 is not the version where we introduced abstract task support, but we only actually need to declare these tasks as `abstract` from Gradle 8.0 so this is sufficient, at least for this quick-fix PR. We can fine-tune this more in a later PR.